### PR TITLE
Out of jokes changelog update

### PR DESCRIPTION
**Pull request by kevingranade**
#### Summary
None

#### Purpose of change
Update the changelog for the first half of April

#### Describe the solution
tools/generate_changelog.py -f --by-date changelog_04 -t ../repo_token -e 2021-04-15 2021-04-01
tac changelog_04 > changelog_04_rev
tools/merge_summaries.sh changelog_04_rev data/changelog.txt
Then copyedit the results.

#### Additional context
Typos happen, and some of the things I prune might be bigger than I remember or part of some larger set of changes that should go in the changelog, please take a look at not just the final diff, but also the copyedit commit to check for missed entries.

### CODE REVIEW

File: data/changelog.txt [hunk 1]
```diff
@@ -278,6 +278,10 @@ Fixes furniture being 'destroyed' when grab-moving.
 Fixes crashes from canceling reloading a container that is on the ground.
 Fixes NPCs still being hungry - and unhappy about it - after eating from faction camp stores.
 Reined in plant mutation photosynthesis ability to be more representative of real photosynthesis.
+Prevents cross-z-level melee attacks when attacker or target is not actually visible.
+Do not do character body updates more than once per turn
+When harvesting crops, drop the result on player's square.
+Improves road system connectivity across Overmaps.
```
Comment (actual-nh) [inline data/changelog.txt:284]:
```suggestion
Prevents cross-z-level melee attacks when the target is not actually visible.
Avoids doing character body updates more than once per turn
When harvesting crops, drops the result on the player's square.
Improves road system connectivity across overmaps.
```

**Review by actual-nh — COMMENTED** [review]
Understand!

File: data/changelog.txt [hunk 2]
```diff
@@ -278,6 +278,10 @@ Fixes furniture being 'destroyed' when grab-moving.
 Fixes crashes from canceling reloading a container that is on the ground.
 Fixes NPCs still being hungry - and unhappy about it - after eating from faction camp stores.
 Reined in plant mutation photosynthesis ability to be more representative of real photosynthesis.
```
Comment (actual-nh) [inline data/changelog.txt:280]:
```suggestion
Reins in the plant mutation category's photosynthesis ability to be more representative of real photosynthesis.
```

File: data/changelog.txt [hunk 3]
```diff
@@ -378,6 +382,7 @@ Improves scooter tile representation by adding scooter vehicle parts, which scoo
 Reduces file size of composed tilesheets from compose.py.
 Implements C++ symbol demangling in crash reports on MacOS, MinGW, and Linux.
```
Comment (BrettDong) [inline data/changelog.txt:379]:
```suggestion
 Implements C++ symbol demangling in crash reports on MinGW, Linux, macOS, BSD and Android.
```

Comment (actual-nh) [reply data/changelog.txt:379]:
```suggestion
Implements C++ symbol demangling for crash reports on MinGW, Linux, macOS, BSD, and Android.
```

**Pull request merged**